### PR TITLE
DOCS-3084: Update agent.md

### PR DESCRIPTION
**Pull request by npentrel**
@benjirewis - thoughts on this? Does that cover it?
https://docs-test.viam.dev/3726/configure/agent/#version-updates

### CODE REVIEW

File: docs/configure/agent.md [hunk 1]
```diff
@@ -188,6 +188,11 @@ Edit and fill in the attributes as applicable.
 | `disable_subsystem` | boolean | Optional | When set to `true` it disables the management of `viam-server`. |
 | `attributes` | object | Optional | <ul><li>`fast_start`: If set to `true`, `viam-agent` will not wait for a network connection nor check for updates before starting `viam-server`. See [Fast start mode](#fast-start-mode).</li></ul> |
 
+### Version updates
```
Comment (npentrel) [inline docs/configure/agent.md:191]:
```suggestion
#### Version updates
```

File: docs/configure/agent.md [hunk 2]
```diff
@@ -188,6 +188,11 @@ Edit and fill in the attributes as applicable.
 | `disable_subsystem` | boolean | Optional | When set to `true` it disables the management of `viam-server`. |
 | `attributes` | object | Optional | <ul><li>`fast_start`: If set to `true`, `viam-agent` will not wait for a network connection nor check for updates before starting `viam-server`. See [Fast start mode](#fast-start-mode).</li></ul> |
 
+#### Version updates
+
+To avoid unexpected downtime when `viam-server` is updated, you can configure a [Maintenance Window](/architecture/viam-server/#maintenance-window).
+With a configured maintennce window, `viam-agent` will restart and upgrade `viam-server` only when maintenance is allowed and when `viam-server` is not currently reconfiguring resources.
```
Comment (benjirewis) [inline docs/configure/agent.md:194]:
```suggestion
With a configured maintenance window, `viam-agent` will restart and upgrade `viam-server` only when maintenance is allowed and when `viam-server` is not currently reconfiguring resources.
```

Comment (benjirewis) [inline docs/configure/agent.md:194]: > not currently reconfiguring resources

NetCode uses the word "reconfiguring" all the time, but do users know what this means? Should we say "processing config changes" or something more generic? Up to you was just curious myself.

**Review by benjirewis — APPROVED** [review]
Pretty much LGTM.

File: docs/configure/agent.md (hunk 2 above)
Comment (npentrel) [reply docs/configure/agent.md:194]:
```suggestion
With a configured maintenace window, `viam-agent` will restart and upgrade `viam-server` only when maintenance is allowed and when `viam-server` is not currently processing config changes.
```

File: docs/configure/agent.md [hunk 3]
```diff
@@ -188,6 +188,11 @@ Edit and fill in the attributes as applicable.
 | `disable_subsystem` | boolean | Optional | When set to `true` it disables the management of `viam-server`. |
 | `attributes` | object | Optional | <ul><li>`fast_start`: If set to `true`, `viam-agent` will not wait for a network connection nor check for updates before starting `viam-server`. See [Fast start mode](#fast-start-mode).</li></ul> |
 
+#### Version updates
+
+To avoid unexpected downtime when `viam-server` is updated, you can configure a [Maintenance Window](/architecture/viam-server/#maintenance-window).
+With a configured maintenace window, `viam-agent` will restart and upgrade `viam-server` only when maintenance is allowed and when `viam-server` is not currently processing config changes.
```
Comment (npentrel) [inline docs/configure/agent.md:194]:
```suggestion
With a configured maintenance window, `viam-agent` will restart and upgrade `viam-server` only when maintenance is allowed and when `viam-server` is not currently processing config changes.
```

**Pull request merged**